aws/resource_aws_sfn_state_machine_test.go

```diff
@@ -129,38 +173,14 @@ func testAccCheckAWSSfnStateMachineDestroy(s *terraform.State) error {
 			return fmt.Errorf("Expected AWS Step Function State Machine to be destroyed, but was still found")
 		}
 
-		return nil
+		return err
```

Comment:
Since `err` is already handled, this should be removed so the loop can continue.

```suggestion
```